test/e2e/soak_k8petstore.go

```diff
@@ -0,0 +1,171 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"io/ioutil"
+	"time"
+	//"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+)
+
+var root0 = absOrDie(filepath.Clean(filepath.Join(path.Base(os.Args[0]), "..")))
+var err error = nil
+var namespace *api.Namespace = nil
+var ns string = ""
+
+//This must run on a machine which is in the kube-proxy ring.  Otherwise, the publicIP binding will fail.
+//The IP Below ~ letter->number cipher for k8petstore (most likely it won't be bound by any other process)
+var ip = "165.201.92.15"
+
+// i.e. after 50 trials, expect 3000 transactions... minimum we settle for is 500.
+var minionCount int
+
+// readTransactions reads # of transactions from the k8petstore web server endpoint.
+// for more details see the source of the k8petstore web server.
+func readTransactions(c *client.Client) int {
+
+	resp, err := http.Get("http://165.201.92.15:3000/llen")
+	if err != nil {
+		log.Fatal("FAILED %v", err) // handle error
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	Expect(err).NotTo(HaveOccurred())
+	total_trans, err := strconv.Atoi(string(body))
+	Expect(err).NotTo(HaveOccurred())
+	return total_trans
+}
+
+// runk8petstore runs the k8petstore application, bound to external ip "ip", and polls it to assert that "min_expected"
+// transactions are acquired in a maximum of "max_seconds".
+func runK8petstore(ip string, rest_servers int, load_generators int, c *client.Client, min_expected int, max_seconds int64) {
+
+	k8bps := filepath.Join(root0, "examples/k8petstore/k8petstore.sh")
+
+	//Get the count of minions.  We'll use this to decide expected throughput and loadgen/REST server count.
+	cmd := exec.Command(
+		k8bps,
+		"kubectl", //for dev, replace w/ "cluster/kubectl.sh"
+		"r.2.8.19",
+		ip,
+		"50", // allow max of 50 trials before we expect the app to be running.
+		strconv.Itoa(rest_servers),
+		strconv.Itoa(load_generators),
+		"1", "0", ns) // 1= # slave, 0 = don't bother running the embedded test.
+
+	var out bytes.Buffer
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		fmt.Printf(out.String())
```

Comment:
you don't appear to write anything to out before printing this, so this print is not needed?